req2flatpak.py

```diff
@@ -705,10 +728,31 @@ def main():
     parser = cli_parser()
     options = parser.parse_args()
 
+    def _get_yaml_module_or_exit():
+        try:
+            # optional dependency, not imported at top
+            import yaml
+        except ImportError:
+            parser.error(
+                "Outputing YAML requires 'pyyaml' package: try 'pip install pyyaml'"
+        )
+        return yaml
+
     # stream output to a file or to stdout
-    output_stream = options.outfile if hasattr(options.outfile, "write") else sys.stdout
+    if hasattr(options.outfile, "write"):
+        output_stream = options.outfile
+        if pathlib.Path(output_stream.name).suffix.casefold() in (".yaml", ".yml"):
+            options.yaml = True
+    else:
+        output_stream = sys.stdout
 
     # print platform info if requested, and exit
+    if options.platform_info and options.yaml:
+        yaml = _get_yaml_module_or_exit()
+        yaml.dump(
+            asdict(PlatformFactory.from_current_interpreter()), output_stream, indent=2
+        )
+        parser.exit()
```

Comment:
Not sure whether to value code simplicity over redundancy. I would prioritize the latter and use a nested if-statement.